test/common/testlib.py

```diff
@@ -490,38 +497,45 @@ def run(self, result=None):
             if stopTestRun is not None:
                 stopTestRun()
 
-    def setUp(self, macaddr=None, memory_mb=None, cpus=None):
+    def setUp(self):
+        if opts.address and self.provision is not None:
+            raise unittest.SkipTest("Cannot provision multiple machines if a specific machine address is specified")
```

Comment:
IMHO that would just silently paper over errors in our tests. This should rather be a `ValueError` or other fatal exception.